simson/common/data_extrapolations.py

```diff
@@ -33,29 +26,42 @@ def regress(self):
         pass
 
 
+class OneDimensionalExtrapolation(Extrapolation):
+    @model_validator(mode="after")
+    def validate_data(self):
+        assert self.data_to_extrapolate.ndim == 1, "Data to extrapolate must be 1-dimensional."
+        assert self.target_range.ndim == 1, "Target range must be 1-dimensional."
+        assert self.data_to_extrapolate.shape[0] < self.target_range.shape[0], (
+            "data_to_extrapolate must be smaller then target_range")
+
+
 class WeightedProportionalExtrapolation(Extrapolation):
     """
     Regression of a function of the form y = a * x, i.e. a linear scaling without offset.
     For regression, the last n_last_points_to_match points are used. Their weights are linearly decreasing to zero.
     """
 
+    @model_validator(mode="after")
+    def validate_data(self):
```

Comment:
this should get some additional checks on the shape of the input arrays, I suppose?